typing/ctype.ml

```diff
@@ -2466,30 +2469,23 @@ let find_lowest_level ty =
     end
   in find ty; unmark_type ty; !lowest
 
-let add_gadt_equation env source destination =
+let add_gadt_equation uenv source destination =
```

Comment:
Could you add a comment saying this can be called only in `Pattern` mode? Perhaps this is obvious from the name of the function, but I still think stating preconditions like this help newcomers to the codebase.